### ISSUE TITLE
[BUG] _Different result between `dp --pt test` and `DeepEval` for a fine-tuned multi-task model

### ISSUE THREAD

**Issue by SchrodingersCattt** [deepmodeling/deepmd-kit#3892]
### Bug summary

I finetuned a model based on pre-trained model of [2024Q1 from aissq](https://www.aissquare.com/models/detail?pageType=models&name=DPA-2.1.0-2024Q1&id=244), with `branch` of `Domains_OC2M`. My new datasets are labeled by CP2K 2024.1 (units have been correctedly converted to eV for energies and eV/ang. for forces) for my experimentally-observed structure. The command for finetune is :
```bash
nohup dp --pt train train.json --finetune OpenLAM_2.1.0_27heads_2024Q1.pt --model-branch Domains_OC2M> log 2>&1 && dp --pt freeze 
```

Via `dp --pt test` command, I got good consistency between DFT labels and DP predictions:
![img_v3_02c3_5c8188d5-5fb8-4d0e-ad9b-a62ec0c5d48g](https://github.com/deepmodeling/deepmd-kit/assets/96342692/102c7e96-f913-4a36-a0a5-eda75431530e)

However, I noticed a significant discrepancy between the relabeled data and the original DFT-labeled data with Python API using `DeepEval` module.
![img_v3_02c3_2380fc34-8850-4caf-a89f-5dc97ebff65g](https://github.com/deepmodeling/deepmd-kit/assets/96342692/304afc9b-07a3-40ae-bd4d-8788a9645bbe)

I have attached my datasets, models, input files, and relevant scripts. Please review them if needed.

Thank you for your time!

### DeePMD-kit Version

DeePMD-kit v3.0.0a1.dev82+geed7c8ac

### Backend and its version

torch 2.0.0+cu118

### How did you download the software?

Offline packages

### Input Files, Running Commands, Error Log, etc.

In [dataset.zip](https://github.com/user-attachments/files/15930797/dataset.zip), folder `train` and `valid` are DFT-labeled, while `train_relabeled` and `valid_relabeled` are finetuned model-relabeled.

In [scripts.zip](https://github.com/user-attachments/files/15930895/scripts.zip), `relabel.py` was executed after freezing the model; `plot_test.py` was executed after `dp --pt test`; `plot_relabel.py` was executed after relabel. ( See "Steps to Reproduce" for details)

[train.json](https://github.com/user-attachments/files/15931695/train.json) is my input.

My finetuned model can be accessed here (uploaded to my repo due to the size limit...):
[https://github.com/SchrodingersCattt/deepmd-issue-20240622/blob/main/frozen_model.pth](url)

### Result of `dp --pt test`

Run `dp --pt test  -m frozen_model.pth -s train -d test` then run `python plot_test.py`. This should generate an `e_f.png`.

### Result of `DeepEval` 

Run `python relabel.py` then run `python plot_relabel.py`. This should generate a `dft_vs_relabel.png`.

**Comment by njzjz** [comment]
Duplicate of #3884

**Comment by SchrodingersCattt** [comment]
I finally solved the problem and share my experience here.

The `atype` of model obtained from multi-task training should be the same with the pre-training, i.e. the 118-element `type_map`, INSTEAD OF the `type_map` wrote in `input.json`. That's why I run my scripts all well in single-task pretrained models but fail here in multi-task pretrained models !
